lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -3359,6 +3360,16 @@ void ByteCodeGenerator::EmitOneFunction(ParseNode *pnode)
             }
         }
 
+        // If the function has non simple parameter list, the params needs to be evaluated when the generator object is created
+        // (that is when the function is called). This yield opcode is to mark the  begining of the function body.
```

Comment:
Looks good. Could you just add a todo comment here to confirm that there's no perf hit from the additional next call? (Thanks.)